### PR TITLE
Update the changelog to the latest format

### PR DESCRIPTION
**Pull request by DJMcNab**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,12 +27,28 @@ Subheadings to categorize changes are `added, changed, deprecated, removed, fixe
...
+
+[#26]: https://github.com/linebender/peniko/pull/26
+
+[@ratmice]: https://github.com/ratmice
+[@DJMcNab]: https://github.com/DJMcNab
+[#40]: https://github.com/linebender/peniko/pull/40
+
+[Unreleased]: https://github.com/linebender/peniko/compare/v0.1.1...HEAD
+[0.1.1]: https://github.com/linebender/peniko/compare/v0.1.0...v0.1.1
+[0.1.0]: https://github.com/linebender/peniko/releases/tag/v0.1.0
```
Comment (Philipp-M) [inline CHANGELOG.md:54]: Is this mostly to be able to use that trick for docs.rs? Or for readability?

**Review by Philipp-M — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 1 above)
Comment (DJMcNab) [reply CHANGELOG.md:54]: Can you clarify your question? This is just mindlessly applying the pattern of linebender/vello#604.

The changelog isn't included in rustdoc (at least not currently)

Comment (Philipp-M) [reply CHANGELOG.md:54]: I mean link references vs in place links, but I see that this is just the general format there

**Pull request merged**